phoenix-core/src/main/java/org/apache/phoenix/execute/BaseQueryPlan.java

```diff
@@ -204,8 +204,18 @@ public final ResultIterator iterator(final List<? extends SQLCloseable> dependen
         } else {
             ScanUtil.setTimeRange(scan, context.getScanTimeRange());
         }
-        
-        ScanUtil.setTenantId(scan, connection.getTenantId() == null ? null : connection.getTenantId().getBytes());
+        byte[] tenantIdBytes;
```

Comment:
This is for a case where you have a tenant-specific connection set up and are accessing a table that is not multi-tenant, but it has a different type than your tenant ids. Without this check, the tenantId fails to convert and would prevent us from reading the table even though it is unrestricted.